src/sidebar/util/observe-element-size.js

```diff
@@ -0,0 +1,65 @@
+'use strict';
+
+/**
+ * Watch for changes in the size (`clientWidth` and `clientHeight`) of
+ * an element.
+ *
+ * Returns a cleanup function which should be called to remove observers when
+ * updates are no longer needed.
+ *
+ * @param {Element} element - HTML element to watch
+ * @param {(width: number, height: number) => any} onSizeChanged -
+ *   Callback to invoke with the `clientWidth` and `clientHeight` of the
+ *   element when a change in its size is detected.
+ * @return {() => void}
+ */
+function observeElementSize(element, onSizeChanged) {
+  if (typeof ResizeObserver !== 'undefined') {
+    const observer = new ResizeObserver(() =>
+      onSizeChanged(element.clientWidth, element.clientHeight)
+    );
+    observer.observe(element);
+    return () => observer.disconnect();
+  }
+
+  // Fallback method which listens for the most common events that result in
+  // element size changes:
+  //
+  // - Window size change
+  // - Media loading and adjusting size to content
+  // - DOM changes
+  //
+  // This is not comprehensive but it is simple to implement and good-enough for
+  // our current use cases.
+
+  let prevWidth = element.clientWidth;
+  let prevHeight = element.clientHeight;
+
+  const check = () => {
+    if (
+      prevWidth !== element.clientWidth ||
+      prevHeight !== element.clientHeight
+    ) {
+      prevWidth = element.clientWidth;
+      prevHeight = element.clientHeight;
+      onSizeChanged(prevWidth, prevHeight);
+    }
+  };
+
+  element.addEventListener('load', check);
+  window.addEventListener('resize', check);
+  const observer = new MutationObserver(check);
+  observer.observe(element, {
```

Comment:
It can be a fine line of figuring out when to introduce comments for external APIs, etc., but I'm wondering if a quick comment here would help for future devs in terms of which `MutationObserverInit` properties have been configured here. If I am parsing this correctly, it's something like "Whenever text content changes, child nodes are added/removed or child-node text content changes"?

Reply:
> If I am parsing this correctly, it's something like "Whenever text content changes, child nodes are added/removed or child-node text content changes"?

That is exactly right. I honestly think a comment would add noise since this is straightforward use of the API and MDN [has good documentation](https://developer.mozilla.org/en-US/docs/Web/API/MutationObserver).